TMDS_3/include/CGAL/tetrahedron_soup_to_triangulation_3.h

```diff
@@ -0,0 +1,170 @@
+// Copyright (c) 2009-2014 INRIA Sophia-Antipolis (France).
+// Copyright (c) 2010-2013 GeometryFactory Sarl (France).
+// All rights reserved.
+//
+// This file is part of CGAL (www.cgal.org).
+//
+// $URL$
+// $Id$
+// SPDX-License-Identifier: GPL-3.0-or-later OR LicenseRef-Commercial
+//
+//
+// Author(s)     :  Mael Rouxel-Labbé, Maxime Gimeno, Jane Tournois
+//
+//******************************************************************************
+// File Description :
+//******************************************************************************
+
+#ifndef CGAL_TMDS_3_TETRAHEDRON_SOUP_TO_C3T3_H
+#define CGAL_TMDS_3_TETRAHEDRON_SOUP_TO_C3T3_H
+
+#include <CGAL/license/TMDS_3.h>
+
+#include <CGAL/TMDS_3/tet_soup_to_c3t3.h>
+#include <CGAL/TMDS_3/internal/TMDS_3_helper.h>
+#include <CGAL/boost/graph/named_params_helper.h>
+
+#include <vector>
+#include <array>
+#include <boost/unordered_map.hpp>
+
+namespace CGAL {
+
+  /** \ingroup PkgTMDS3Functions
+  * builds a 3D triangulation from a soup of tetrahedra.
+  *
+  * @tparam TetrahedronRange a model of `Range` whose value type is
+  * a `Tetrahedron_3`
+  * @tparam Triangulation a 3D triangulation class that has
+  * a vertex base model of `MeshVertexBase_3`
+  * and a cell base model of `MeshCellBase_3`
+  *
+  * @param tets each element in the range is the geometric description of the
+  * corresponding cell in `tr`
+  * @param tr the 3D triangulation to be built
+  *
+  * @post the output triangulation must be a triangulation of the convex hull of `tets`
+  */
+  template<typename TetrahedronRange, typename Triangulation>
+  void tetrahedron_soup_to_triangulation_3(const TetrahedronRange& tets,
+                                           Triangulation& tr)
```

Comment:
There is an important precondition for that function, that is not documented: the range `tets` should be the set of finite tetrahedra of a valid CGAL triangulation.